### ISSUE TITLE
OIDC: Back-Channel Logout Request failed

### ISSUE THREAD

**Issue by marabooapp** [keycloak/keycloak#21550]
### Before reporting an issue

- [X] I have searched existing issues
- [x] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

identity-brokering

### Describe the bug

Keycloak Backchannel Logout try to external IdP failed with logged message "Failed backchannel broker logout to: `URL for external IdP`" if ext. IdP expects `HTTP POST` request.

### Version

20.0.5

### Expected behavior

Supporting `HTTP POST` request type for the Backchannel logout from Keycloak to external IdP.

### Actual behavior

In Keycloak, while it has set "OpenID Connect v1.0" Identity provider and "Backchannel logout" is checked then Keycloak is sending only `HTTP GET` request to external IdP for the Backchannel logout, but OIDC specification describes Backchannel logout request as `HTTP POST` type (https://openid.net/specs/openid-connect-backchannel-1_0.html#BCRequest).

It seems to be problem in Keycloak class OIDCIdentityProvider.java which only makes the `HTTP GET` request type (https://github.com/keycloak/keycloak/blob/561bd65b4a55c947c9a891d16353d30c6049da49/services/src/main/java/org/keycloak/broker/oidc/OIDCIdentityProvider.java#L144C1-L154C75)

### How to Reproduce?

Set "OpenID Connect v1.0" Identity Provider which expects `HTTP POST` request within Backchannel Logout.

### Anything else?

Behaviour is checked in Keycloak version 20.0.5 but responsible code is the same in the latest version (21.1.2) too.

**Comment by sschu** [comment]
The code you mentioned is the one for the proprietary Keycloak to Keycloak Backchannel Logout. The official OIDC Logout is here https://github.com/keycloak/keycloak/blob/7cff8572387e6bbfca160aa60e9a8191d22e5c85/services/src/main/java/org/keycloak/services/managers/ResourceAdminManager.java#L211 and it is correctly using a POST request. So I guess it is something in your configuration that is causing the wrong method being used.

**Comment by mposolda** [comment]
@sschu Thanks for your comment!

I am closing as this looks like a misconfiguration rather than Keycloak issue.

**Comment by marabooapp** [comment]
I checked and it is used the configuration "OpenID Connect v1.0" ("oidc" PROVIDER_ID in database) and not "Keycloak OpenID Connect".

In WARN log with error message `Failed backchannel broker logout to:`
- `[org.keycloak.models.AbstractKeycloakTransaction]` in Keycloak version 20.0.5,
- `[org.keycloak.broker.oidc.OIDCIdentityProvider]` in Keycloak version 15.0.2.


I found WARN message `Failed backchannel broker logout to:` only in class OIDCIdentityProvider.java in source code.


There is someting what I missed in configuration if "OpenID Connect v1.0" is acted like "Keycloak OpenID Connect" provider?
